### PR TITLE
E2E: Add option to deploy ceph-csi in different namespace

### PR DESCRIPTION
**Pull request by Madhu-1**
This PR has changes to deploy ceph-csi in a different namespace and also provides an option to enable/disable plugin installation in E2E, which also helps in enabling helm templates testing in CI

fixes: #723

### CODE REVIEW

**Comment by Madhu-1** [comment]
Facing some issues with vault E2E. looking at it

File: e2e/README.md [hunk 1]
```diff
@@ -70,12 +70,16 @@ $./minikube.sh clean
...
 
-| flag           | description                                                                   |
-| -------------- | ----------------------------------------------------------------------------- |
-| deploy-timeout | Timeout to wait for created kubernetes resources (default: 10)                |
-| kubeconfig     | Path to kubeconfig containing embedded authinfo (default: $HOME/.kube/config) |
-| timeout        | Panic test binary after duration d (default 0, timeout disabled)              |
-| v              | Verbose: print additional output                                              |
+| flag              | description                                                                   |
+| ----------------- | ----------------------------------------------------------------------------- |
+| deploy-timeout    | Timeout to wait for created kubernetes resources (default: 10)                |
```
Comment (humblec) [inline e2e/README.md]: Its better to specify the unit of time too.

File: e2e/cephfs.go [hunk 2]
```diff
@@ -25,42 +25,86 @@ var (
...
+	data, err = replaceNamespaceInTemplate(cephfsDirPath + cephfsProvisionerPSP)
 	if err != nil {
-		e2elog.Logf("failed to delete cephfs nodeplugin rbac %v", err)
+		e2elog.Logf("failed to read content from %s %v", cephfsDirPath+cephfsProvisionerPSP, err)
 	}
-	_, err = framework.RunKubectl("delete", "-f", cephfsDirPath+cephfsNodePluginPSP)
+	_, err = framework.RunKubectlInput(data, action, ns, "-f", "-")
 	if err != nil {
-		e2elog.Logf("failed to delete cephfs nodeplugin psp %v", err)
+		e2elog.Logf("failed to %s cephfs provisioner psp %v", action, err)
```
Comment (humblec) [inline e2e/cephfs.go]: better to expand `psp` in above logs.

File: e2e/cephfs.go [hunk 3]
```diff
@@ -70,8 +114,17 @@ var _ = Describe("cephfs", func() {
 	// deploy cephfs CSI
 	BeforeEach(func() {
 		c = f.ClientSet
+		if deployCephFS {
+			if cephCSINamespace != defaultNs {
+				// create namespace
+				err := createNamespace(c, cephCSINamespace)
```
Comment (humblec) [inline e2e/cephfs.go]: Should be fine to drop above source code comment too  as its pretty evident.

File: e2e/rbd.go [hunk 4]
```diff
@@ -23,47 +23,88 @@ var (
...
+
+	createORDeleteRbdResouces("create")
 }
 
 func deleteRBDPlugin() {
-	_, err := framework.RunKubectl("delete", "-f", rbdDirPath+rbdProvisioner)
+	createORDeleteRbdResouces("delete")
+}
+
+func createORDeleteRbdResouces(action string) {
```
Comment (humblec) [inline e2e/rbd.go]: somehow I feel above func name is not great :) , we could think some more better name if possible

File: scripts/travis-functest.sh [hunk 5]
```diff
@@ -12,6 +12,6 @@ sudo scripts/minikube.sh k8s-sidecar
 sudo chown -R travis: "$HOME"/.minikube /usr/local/bin/kubectl
 # functional tests
 
-go test github.com/ceph/ceph-csi/e2e --deploy-timeout=10 -timeout=30m -v
+go test github.com/ceph/ceph-csi/e2e --deploy-timeout=10 -timeout=30m --cephcsi-namespace=cephcsi-testing -v
```
Comment (humblec) [inline scripts/travis-functest.sh]: may be a variable for value `cephcsi-testing` is better.

**Comment by humblec** [comment]
@Madhu-1 few comments, also its better to change the title to reflect this is for e2e .

File: e2e/cephfs.go (hunk 2 above)
Comment (Madhu-1) [reply e2e/cephfs.go]: `psp` is not added in this PR, if required we can do a cleanup in follow up PR

File: e2e/rbd.go (hunk 4 above)
Comment (Madhu-1) [reply e2e/rbd.go]: can you please suggest a new name for it?

File: scripts/travis-functest.sh (hunk 5 above)
Comment (Madhu-1) [reply scripts/travis-functest.sh]: i don't think we need to define a new variable and use it, I have just followed a pattern which we used earlier.

**Comment by humblec** [comment]
Thanks @Madhu-1 for addressing the comments.  LGTM.  Approved.

**Comment by humblec** [comment]
@nixpanic can you review this PR please ?

**Comment by humblec** [comment]
@nixpanic ping ? :)

**Comment by humblec** [comment]
@nixpanic @ShyamsundarR any review comments here?

File: scripts/travis-functest.sh (hunk 5 above)
Comment (nixpanic) [reply scripts/travis-functest.sh]: I prefer to use a changing namespace, something like `--cephcsi-namespace=cephcsi-testing-${RANDOM}`. That makes sure the namespace will not get hardcoded in other locations later on.

**Comment by Madhu-1** [comment]
@nixpanic updated. @humblec @ShyamsundarR PTAL

**Comment by Madhu-1** [comment]
@MergifyIO  rebase

**Comment by humblec** [comment]
LGTM. Thanks.. Approved.

**Comment by humblec** [comment]
@nixpanic ptal.

**Pull request merged**